src/Codec/FFmpeg/Encode.hsc

```diff
@@ -95,44 +97,57 @@ foreign import ccall "av_image_fill_arrays"
 foreign import ccall "av_image_fill_linesizes"
   av_image_fill_linesizes :: Ptr CInt -> AVPixelFormat -> CInt -> IO CInt
 
+foreign import ccall "av_frame_make_writable"
+  av_frame_make_writable :: AVFrame -> IO CInt
+
 -- * FFmpeg Encoding Interface
 
 -- | Minimal parameters describing the desired video output.
 data EncodingParams =
```

Comment:
So I moved any video specific fields to a separate record called `VideoParams` below and less top level specific params in this record.